src/Test.idr

```diff
@@ -0,0 +1,57 @@
+module Test
+
+import Data.Vect 
+
+import Typedefs
+import Types
+import Backend.Utils
+
+import Specdris.Spec
+import Text.PrettyPrint.WL
+
+%access public export
+
+bit : TDef 0
+bit = TName "Bit" $ TSum [T1, T1]
+
+byte : TDef 0
+byte = TName "Byte" $ pow 8 bit
+
+maybe : TDef 1
+maybe = TName "Maybe" $  TSum [T1, TVar 0]
+
+list : TDef 1
+list = TMu "List" [("Nil", T1), ("Cons", TProd [TVar 1, TVar 0])]
+
+maybe2 : TDef 1
+maybe2 = TMu "Maybe2" [("Nothing", T1), ("Just", TVar 1)]
+
+nat : TDef 0
+nat = TMu "Nat" [("Z", T1), ("S", TVar 0)]
+
+listNat : TDef 0 
+listNat = TMu "ListNat" [("NilN", T1), ("ConsN", TProd [weakenTDef nat 1 LTEZero, TVar 0])]
+
+parametricSynonym : TDef 1
+parametricSynonym = TName "ParSyn" maybe
+
+parametricSynonym2 : TDef 1
+parametricSynonym2 = TName "ParSyn2" maybe2
+
+aplusbpluscplusd : TDef 4
+aplusbpluscplusd = TName "aplusbpluscplusd" $ TSum [TVar 0, TVar 1, TVar 2, TVar 3]
+
+oneoneoneone : TDef 0
+oneoneoneone = TName "oneoneoneone" $ TSum [T1, T1, T1, T1]
+
+listByte : TDef 0
+listByte = TMu "ListByte" [("NilC", T1), ("ConsC", TProd [weakenTDef byte 1 LTEZero, TVar 0])]
+
+charlistNatalphaTriples : TDef 1
```

Comment:
Could this name be made a bit clearer? I think capitalising it as `charListnatAlpha` would already help a bit.

Reply:
Changed it to `tripleByteListnatAlpha`, in keeping with the naming convention where the parametric type comes before the parameters. Good enough?

Reply:
I take the :heart: as an approval. :)